AstSemantics.md

```diff
@@ -111,6 +112,9 @@ Integer loads can specify a *storage size* which is smaller than the result type
 well as a signedness which determines whether the bytes are sign- or zero-
 extended into the result type.
 
+Each `load` and `store` specifies the [linear memory index](Modules.md#index-spaces)
+to operate on with an unsigned integer immediate.
+
```

Comment:
I agree with the code sharing concern: different dynamic libraries have to agree what linear memory index values mean. They could say `0` is HEAP, `1` is user stack, `2` is rodata, `4` is globals, etc. They need to agree on it, though. Maybe we want to use strings here as well, and somehow make it compress well?

I think we may want to start with @lukewagner's proposal (stick with integers), and then as we get information on compression we can try out different approaches to resolve this problem.